### DISCUSSION TITLE
[PSD10] Exercise 2.43

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/psd2223#168] [Esercizi (A-L)]
![image](https://user-images.githubusercontent.com/69389461/210061000-905c671e-17f5-4071-8b38-13129f017c42.png)

![image](https://user-images.githubusercontent.com/69389461/210061229-26092fee-a417-49e1-a015-bc4e98c9372c.png)

![image](https://user-images.githubusercontent.com/69389461/210061056-0ebbb939-b841-42f3-8326-c9678a3a2b6f.png)

**Comment by fraxb04** [comment]
@Elia-Belli  allego la foto del circuito semplificato con il bubble pushing, e tuttavia viene diverso il risultato:
- tpd:
       short path = 55 ps (1 NOT, 1 OR)
       critical path = 225 ps (5 NOT, 1 AND, 3 OR)
- tcd:
       short path = 40 ps (1 NOT, 1 OR)
       critical path = 165 ps (5 NOT, 1 AND, 3 OR)
![image](https://user-images.githubusercontent.com/122573871/213441135-4e262970-f34b-4ac3-abbc-0b8df26393b0.png)

**Comment by Elia-Belli** [comment]
@fraxb04 il percorso critico è il più lungo, con il Bubble Pushing hai 1 NOT, 2 OR e 1AND nel peggiore dei casi, facendo i calcoli:
$t_{pd}=(15+40* 2+30)= 125ps $
$t_{cd}=(10+30* 2+25)= 95ps$
Te invece stai sommando i delay di tutte le porte che non ha molto senso
**Osservazione**: con il BubblePushing la logica rimane invariata ma il delay è una variabile che dipende dalla porta logica usata e non dalla logica, quindi è normale che cambiandole venga differente

**Comment by fraxb04** [comment]
@Elia-Belli scusami, quindi prendiamo solo il percorso di un input, in questo caso A, poiché se prendessimo tutta la porta (quindi con entrambi gli input) avremmo 2 NOT, giusto?
Inoltre se prendessimo solo lo short path non sarebbe 1 OR ed 1 NOT, quindi 55 ps??

**Comment by Elia-Belli** [comment]
Il percorso critico è indicato dall'input che attraversa più porte per arrivare all'ouput, nel caso di stesso numero di porte si calcolano entrambi i delay e si sceglie il maggiore: ovviamente nel calcolare i delay si prendono solo i delay delle porte attraversate da quell'input e non da altri.
Lo short path è giusto, ma non è necessario ai fini di stabilire i propagation e contamination delays

**Comment by fraxb04** [comment]
@Elia-Belli ok, quindi è giusto 125 ps, mentre il contamination delay è 95 ps.

**Comment by neoSnakex34** [comment]
I risultati riportati sono errati. L'esercizio non richiede di semplificare la rete o altro chiede solo di calcolare i ritardi di propagazione e di contaminazione in critical path e short path.
Dato che abbiamo tutte porte nand a due ingressi il critical path è 3xtpdNAND2 = 60ps.
Lo short path è 1xtcdNAND2 = 15ps.

I risultati sono confermati dalle correzioni ufficiali del libro inglese.

**Reply by niccolo-fato** [reply]
dove si possono trovare le soluzioni ufficiali?
